Rework usage of context.Context in api/client

vdemeester opened the pull request:
Make better default usage on context.Context on the `api/client` package to share the context (it is useless if not shared, which was the case for a lot of commands) 🐮.

I did this pretty naively though and did not think of other possible non default (`context.Background()`) usage expected for the current one (currently `start`, `run`). But at least the changes for those would be easier then.

/cc @stevvooe @aaronlehmann @calavera @runcom @LK4D4 

🐸

_This follows quick discussion on https://github.com/docker/docker/pull/22861#discussion-diff-64094963_

Signed-off-by: Vincent Demeester vincent@sbr.pm

thaJeztah commented:
Flaky test on userns?
https://jenkins.dockerproject.org/job/Docker-PRs-userns/9729/console
```
11:12:57 FAIL: docker_cli_run_test.go:2901: DockerSuite.TestRunContainerWithRmFlagCannotStartContainer
11:12:57 
11:12:57 docker_cli_run_test.go:2914:
11:12:57     c.Fatal("Expected not to have containers", out)
11:12:57 ... Error: Expected not to have containersf0b441933c3f
```

thaJeztah commented:
Actually, may be genuine; failing on win2lin as well https://jenkins.dockerproject.org/job/Docker-PRs-Win2Lin/26090/console

vdemeester commented:
@thaJeztah yep, I might have broke something :sweat:

thaJeztah commented:
Overall, looking good, nice!

vdemeester committed:
Rework usage of context.Context in api/client

Make better default usage on context.Context on the `api/client` package
to share the context (it is useless if not shared, which was the case
for a lot of commands).

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

LK4D4 commented:
LGTM

tiborvass commented:
LGTM
